File: 005.py
"""Find the smallest possible number that is the LCM of 1-20."""

LCM_1_TO_10 = 2520

def divisible_by_one_to_twenty(n):
    """Determine if n is divisible by integers in range 1-20."""
    for i in xrange(20, 10, -1):
        if n % i != 0:
            return False
    if n % LCM_1_TO_10 != 0:
        return False
    return True

num = LCM_1_TO_10 * 2
while True:
    if divisible_by_one_to_twenty(num):
        print num
        break
    else:
        num += LCM_1_TO_10

File: 005.swift
#!/usr/bin/xcrun swift

// Find the smallest possible number that is the LCM of 1-20.

let lcm1To10 = 2_520

func divisibleByOneToTwenty(n: Int) -> Bool {
    // Determine if n is divisible by integers in range 1-20.
    for var i = 20; i >= 10; --i {
        if n % i != 0 {
            return false
        }
    }
    if n % lcm1To10 != 0 {
        return false
    }
    return true
}

var num = lcm1To10 * 2
while true {
    if divisibleByOneToTwenty(num) {
        println(num)
        break
    } else {
        num += lcm1To10
    }
}
